easybuild/framework/easyblock.py

```diff
@@ -594,6 +594,7 @@ def collect_exts_file_info(self, fetch_files=True, verify_checksums=True):
                         if 'source_urls' not in source:
                             source['source_urls'] = source_urls
 
+                        ext_src['sources'] = sources
```

Comment:
Hmm, shouldn't this be `[source]` (to always have a dict value here)?

Otherwise we'll sometimes have a dict value, sometimes a string value, which is annoying to deal with wherever this is used...

Reply:
The idea is that similar to `'patches'` this is always a list of filenames. See the test(s). This is especially important for the `nosource` option.